source/common/stats/metric_impl.cc

```diff
@@ -0,0 +1,73 @@
+#include "common/stats/metric_impl.h"
+
+#include "envoy/stats/tag.h"
+
+#include "common/stats/symbol_table_impl.h"
+
+namespace Envoy {
+namespace Stats {
+
+MetricImpl::~MetricImpl() {
+  // The storage must be cleaned by a subclass of MetricImpl in its
+  // destructor, because the symbol-table is owned by the subclass.
+  // Simply call MetricImpl::clear() in the subclass dtor.
+  ASSERT(!stat_names_.populated());
+}
+
+MetricImpl::MetricImpl(absl::string_view tag_extracted_name, const std::vector<Tag>& tags,
+                       SymbolTable& symbol_table) {
+  // Encode all the names and tags into transient storage so we can count the
+  // required bytes.
+  uint32_t num_names = 1 /* tag_extracted_name */ + 2 * tags.size();
```

Comment:
Could you add more documentation to this line? It's unclear at first why we need 2x the size of the tags.

Reply:
Done -- added more detailed comment above and removed the inline comment.